src/Xrefcheck/Config.hs

```diff
@@ -54,8 +54,86 @@ data NetworkingConfig' f = NetworkingConfig
   , ncMaxRetries                :: Field f Int
     -- ^ How many attempts to retry an external link after getting
     -- a "429 Too Many Requests" response.
+  , ncExternalRefRedirects      :: RedirectConfig
+    -- ^  Rules to override the redirect behavior for external references.
```

Comment:
I would actually make it optional and treat it as if the user said: I don't care about redirects, handle them somehow. And I would default this to `[]`.

And later when we see a redirect that matched no rule, we would treat that as OK, since we have to be optimistic.

What do you think?

Reply:
But then we would be changing the default behavior that we previously implemented. Right? It is now provided with these two default rules:

```
- to: .*
  on: permanent
  outcome: invalid
- to: .*
  on: temporary
  outcome: valid
```

Reply:
Right. I think we really want two things:
1. By default, suggest to the user those defaults you implemented in the previous PR: temporary=ok, permanent=failure.
2. Give the user a simple way to say "don't bother me with redirects".

And I think, making `ncExternalRefRedirects` optional and defaulting it to `[]` that would then be interpreted as "always ok" is the best way to handle the second point.

And IMO insisting on the old behaviour is not so important, different users may find it weird depending on the use cases they face in real life. It is easy to obtain our recommended defaults for config, and I think that's enough.

A made a little detour with my reasoning for this, but this is how I see it now. There are other options to approach all this, but to me `[]` default yet seems to be the winning one.

Reply:
We can use [] for the defaults, but add our previous behaviour as an example in the default config comments.

Reply:
Maybe I'm missing something. If we default `ncExternalRefRedirects` to `[]` (in the default config I guess) then, how do we suggest the previously implemented defaults? With a section in the README file?

Reply:
Hm, maybe we could write it as a comment in the default config, as our "recommendation" :)

Reply:
Under "using `[]` as default" I understood - if `ncExternalRefRedirects` is missing in the config, parse it as if it was `[]` (it really required clarification). And our recommendation for this field will be in the default config.